### DISCUSSION TITLE
How to typecheck whole project

### DISCUSSION THREAD

**Discussion by KotlinIsland** [denoland/deno#9735] [Q&A]
I think it's something like:
`deno test {**,.}/*.ts --no-run`

**Comment by nayeemrmn** [comment]
`deno test --no-run` should do it. Or if for example you run `deno test -A src` to run your tests, it would be `deno test --no-run src`.

**Comment by KotlinIsland** [comment]
```
>deno test --no-run --unstable
No matching test modules found
```
I want to check types on every file, like how fmt/lint check every file.

**Comment by nayeemrmn** [comment]
Does `deno test --unstable` find the files?

**Comment by KotlinIsland** [comment]
```
> deno test --unstable
No matching test modules found
```

There are no test files, I just want to check types on src files.

**Comment by KotlinIsland** [comment]
I would expect `deno lint` to check types tbh

**Comment by kitsonk** [comment]
A lot of the type checking in Deno is situational, for example, scripts imported as web workers are type checked as web workers.  Not having a root module that is run means that valuable information is missing, meaning that the type checking can be inaccurate or incomplete.

What would be a use case for type checking just every file when that file is not being imported anywhere?

**Reply by nayeemrmn** [reply]
@KotlinIsland Yeah, we have `deno test --no-run` to compile every module with any test coverage, but it doesn't appear that you have any test modules. Depending on your project structure, `deno cache <mod.ts or other entrypoint>` could reach everything you want compiled.

Or if you just want globs that match `deno lint` and `deno fmt`, run `deno cache {.,**}.{js,ts,jsx,tsx,mjs,mjsx}`. These globs must be evaluated by your shell.

**Comment by KotlinIsland** [comment]
Thanks, I was just setting up a CI for my project and didn't see an option for checking whole project.

Checking main entry point and utility scripts should be adequate, thanks 🙂.

**Comment by jaydenseric** [comment]
Here is a solution: https://github.com/denoland/deno/discussions/12113#discussioncomment-3214298
